\abstract{%
The genome sequence encodes the blueprint of life, hence, interpreting the genome of different species is one of the leading challenges in genetic and biological research. The main objective of this project is to perform mixture modelling of biological data generated from high-throughput sequencing technology. Initially, by exploiting higher order spatial features of the DNA methylation profiles, we propose a novel approach for modelling methylation profiles around promoter regions. Taking into account the nature of the methylation data produced from high-throughput experiments, we model each promoter region with a set of latent basis functions, using the Binomial distributed Probit regression model, and subsequently EM algorithm is applied to cluster similar methylation profiles. The second contribution of this thesis, is to perform integrative Bayesian analysis of heterogeneous types of high-throughput sequencing data generated from different platforms. We extend the Bayesian Consensus Clustering (BCC) model, which is a flexible statistical method that captures important source-specific features, but simultaneously models shared features between the data sources, to account for different observation models that high-throughput datasets follow. Preliminary results, on synthetic and real datasets, reveal the statistical power of the proposed methods in modelling next-generation sequencing data, and we are confident that they can be applied as everyday analysis tools for bioinformaticians who want to perform exploratory analysis on high-throughput sequencing data. 
}